### ISSUE TITLE
Shader Editor: pow(float, int) works on editor, fails compilation on mobile

### ISSUE THREAD

**Issue by noidexe** [godotengine/godot#25438]
**Godot version:**
3.1.beta.calinou 2e5ede7

**OS/device including version:**
Linux 64, Android

**Issue description:**
I have pow( func_returning_a_float(), 3) in a shader. Pow doesn't actually have a pow(float, int) overload but this shows no warning, and works fine on desktop linux. On andoid, however, compilation will fail with the following error:
```
E 0:00:25:0337   CanvasShaderGLES3: Fragment Program Compilation Failed:
ERROR: 0:383: 'pow' : no matching overloaded function found 
ERROR: 1 compilation errors.  No code generated.


  <C Source>     drivers/gles3/shader_gles3.cpp:173 @ _display_error_with_code()
```
If I change pow( func_returning_a_float(), 3) to pow( func_returning_a_float(), 3.0) it'll work fine.
I guess it should either work on both targets or complain in the editor about wrong parameter types.

**Comment by clayjohn** [comment]
I can confirm that no error is generated on intel in the editor. 

An error should be generated. This is a bug.

**Comment by akien-mga** [comment]
This should be fixed by fd68bb2596bff1990711862f90b4763553d2edac, which made scalar conversions errors on desktop too.

Then ffb9f342a52186b7fb7da4607af524786fea829e implemented implicit casting from int to float, so it would again no longer raise an error, but should work fine on mobile too (where the int will be cast to float). Can you confirm?

**Comment by akien-mga** [comment]
Closing as fixed by ffb9f34, but please comment if you can still reproduce the issue and I'll reopen.
